src/middlewared/middlewared/service.py

```diff
@@ -541,6 +542,17 @@ class ServicePartBase(metaclass=ServicePartBaseMeta):
 
 class CoreService(Service):
 
+    @accepts(Str('id'), Int('cols'), Int('rows'))
+    async def resize_shell(self, id, cols, rows):
```

Comment:
Should this really be a coroutine? Can the input queue put block if its full?

Reply:
It's infinite